test/src/main/java/org/apache/accumulo/test/ShellServerIT.java

```diff
@@ -1431,18 +1436,20 @@ public void help() throws Exception {
     }
   }
 
-  // TODO - evaluate this test is testing what is expected or history is working.
   @Test
   public void history() throws Exception {
-    final String table = name.getMethodName();
+    final String table = getUniqueNames(1)[0];
 
-    ts.exec("history -c", true);
-    ts.exec("createtable " + table);
-    ts.exec("deletetable -f " + table);
-    // TODO - this may be passing only because method name is history.
+    // test clear history command works
+    ts.writeToHistory("foo");
+    ts.exec("history -c", true, "foo", false);
+
+    // write to history file. Does not execute functions
+    ts.writeToHistory("createtable " + table);
```

Comment:
This could verify that the table name and these commands are not present first:

```suggestion
    // write to history file. Does not execute functions
    ts.exec("history", true, table, false);
    ts.exec("history", true, "createtable", false);
    ts.exec("history", true, "deletetable", false);
    ts.writeToHistory("createtable " + table);
```

Reply:
If validity check for the clear works correctly, then I am not sure this step is necessary but I don't have anything against being thorough so I will add.